packages/gatsby-mdx/utils/create-mdx-node.js

```diff
@@ -38,19 +24,19 @@ module.exports = async (
     _PARENT: node.id
   };
 
-  mdxNode.meta = meta;
   mdxNode.excerpt = frontmatter.excerpt;
   mdxNode.exports = nodeExports;
-  mdxNode.rawBody = nodeContent;
+  mdxNode.rawBody = content;
+
+  // Add path to the markdown file path
+  if (node.internal.type === `File`) {
+    mdxNode.fileAbsolutePath = node.absolutePath;
```

Comment:
seems like a `createNodeField` would be better for the file path, wdyt? I recognize that this is probably for API compatibility with transformer-remark but it feels weird to have in the main schema when it's already accessible via the parent node and only accessible for files anyway.

Reply:
When you're creating the node, Gatsby doesn't seem to suggest using `createNodeField` – that is for modifying existing nodes you don't control. Gatsby will throw an error if you try to set the `fields` field on a node you create. The more we can stay inline with the remark plugin, the better. I see no harm in copying this pattern.